language/move-prover/src/bytecode_translator.rs

```diff
@@ -759,6 +805,115 @@ impl<'env> ModuleTranslator<'env> {
                 args
             )
         }
+
+        self.writer.unindent();
+        emitln!(self.writer, "}");
+        emitln!(self.writer);
+    }
+
+    /// Spec check body
+    fn generate_function_spec_check_body(
+        &self,
+        func_target: &FunctionTarget<'_>,
+        spec_check: &SpecCheck,
+    ) {
+        let SpecCheck {
+            id,
+            spec,
+            code,
+            loc: _,
+        } = &spec_check;
+        let conds = &spec.conditions;
+
+        // Find the specification check assertion
+        // Only one is allowed at a time because Boogie is not
+        // gauranteed to return all errors in a procedure
+        let spec_check_opt = conds.iter().find(|cond| match cond.kind {
+            ConditionKind::RequiresSpecCheckAssert | ConditionKind::Ensures => true,
+            _ => false,
+        });
+        let spec_check = spec_check_opt
+            .expect("There should be at least one assertion for the specification check.");
```

Comment:
Is this an error the user can create? Then we must not crash but produce an error. If it is not, how it is prevented that this will fire?